framework/include/auxkernels/AuxKernel.h

```diff
@@ -126,22 +126,26 @@ class AuxKernel :
   SubProblem & _subproblem;
   /// System this kernel is part of
   SystemBase & _sys;
+  /// Nonlinear system, displaced if this kernel is on the auxiliary displaced system
   SystemBase & _nl_sys;
+  /// Auxiliary undisplaced system
   AuxiliarySystem & _aux_sys;
   /// Thread ID
   THREAD_ID _tid;
   /// Assembly class
   Assembly & _assembly;
   /// Variable this kernel is acting on
   MooseVariable & _var;
+  /// Displaced problem (valid only when it exists)
+  MooseSharedPointer<DisplacedProblem> _displaced_problem;
```

Comment:
@permcody I'm starting to see the problem here.  It's a bad idea for objects like AuxKernels to have shared pointers to objects like the Systems.  It doesn't make sense.  AuxKernel doesn't have shared ownership of the DisplacedProblem...

@YaqiWang I'm not saying anything against this code... you're doing the "right" thing... the interface returns a shared pointer and that's how you're catching it...